src/Common/DNSResolver.cpp

```diff
@@ -142,6 +142,9 @@ static String reverseResolveImpl(const Poco::Net::IPAddress & address)
 
 struct DNSResolver::Impl
 {
+    using HostWithConsecutiveFailures = std::unordered_map<String, UInt8>;
+    using AddressWithConsecutiveFailures = std::unordered_map<Poco::Net::IPAddress, UInt8>;
```

Comment:
Should we throw an exception if `dns_max_consecutive_failures` is greater than 255?

Reply:
Great point. Initially I was using a static 5 so I used UInt8 but this should be a UInt32 to match the config. Then there is no need to throw